scalding-core/src/main/scala/com/twitter/scalding/reducer_estimation/Common.scala

```diff
@@ -55,24 +55,34 @@ object Common {
     unrollTaps(step.getSources.asScala.toSeq)
 
   /**
-   * Get the total size of the input paths, which may contain a glob
+   * Get the total size of the file(s) specified by the Hfs, which may contain a glob
    * pattern in its path, so we must be ready to handle that case.
+   *
+   * As first step we ask size of tap, if we fail or tap return size is 0
+   * we try to calculate an size by ourself
    */
-  def inputSizes(step: FlowStep[JobConf]): Seq[(Path, Long)] = {
-    val conf = step.getConfig
-
-    FileInputFormat
-      .getInputPaths(conf)
-      .map { path =>
-        val fs = path.getFileSystem(conf)
-        val size = fs.globStatus(path)
-          .map(status => fs.getContentSummary(status.getPath).getLength)
-          .sum
-
-        path -> size
+  def size(tap: Hfs, conf: JobConf): Option[Long] = {
+    val tapSize = Try { tap.getSize(conf) }.toOption
```

Comment:
I know the case when we can calculate by ourself, because generic Hfs getSize doesn't handle paths glob.